Source/RapyutaSimulationPlugins/Public/Core/RRMathUtils.h

```diff
@@ -96,6 +97,34 @@ class RAPYUTASIMULATIONPLUGINS_API URRMathUtils : public UBlueprintFunctionLibra
         return false;
     }
 
+    /**
+     * @brief Clamp a given rotator to its max axis angles
+     *
+     * @param InRotator
+     * @param InMaxAngles
+     * @return Clamped rotator
+     */
+    FORCEINLINE static void ClampRotatorToMaxAngles(FRotator& InRotator, const FRotator& InMaxAngles)
+    {
+        InRotator = FRotator(ClampAngle(InRotator.Pitch, InMaxAngles.Pitch),
+                             ClampAngle(InRotator.Yaw, InMaxAngles.Yaw),
+                             ClampAngle(InRotator.Roll, InMaxAngles.Roll));
+    }
+
+    /**
+     * @brief Clamp an angle in a range [-InMaxAxisAngle, InMaxAxisAngle]
+     * @tparam T
+     * @param InAngle
+     * @param InMaxAngle
+     * @return Clamped angle
+     */
+    template<typename T>
+    FORCEINLINE static T ClampAngle(T InAngle, const T InMaxAngle)
+    {
+        T clampedAngle = FRotator::NormalizeAxis(InAngle);
+        return (clampedAngle > InMaxAngle) ? InMaxAngle : (clampedAngle < -InMaxAngle) ? -InMaxAngle : clampedAngle;

```

Comment:
better to use
Clamp
https://docs.unrealengine.com/4.26/en-US/API/Runtime/Core/Math/FMath/Clamp/